internal/npm_install/generate_pnp_build_files.ts

```diff
@@ -0,0 +1,61 @@
+import * as fs from "fs";
+import * as path from "path";
+
+const workspacePath = process.argv.slice(2)[0];
+
+function mkdirp(p: string) {
+  if (!fs.existsSync(p)) {
+    mkdirp(path.dirname(p));
+    fs.mkdirSync(p);
+  }
+}
+
+function writeFileSync(p: string, content: string) {
+  mkdirp(path.dirname(p));
+  fs.writeFileSync(p, content);
+}
+
+function createBuildFile(pkg: string) {
+  const contents = `package(default_visibility = ["//visibility:public"])
+load("@build_bazel_rules_nodejs//internal/npm_install:node_module_library.bzl", "node_module_library")
+
+filegroup(
+name = "${pkg}",
+srcs = []
+)  `;
+
+  writeFileSync(`${pkg}/BUILD.bazel`, contents);
+}
+
+async function processDependencies(
+  deps: string[],
+  type: "dependencies" | "devDependencies"
+): Promise<Set<string>> {
+  const flattenedDeps: Set<string> = new Set();
+  deps.forEach((key: string) => {
+    // loop over transitive dependencies here
+    flattenedDeps.add(key);
+  });
+  return flattenedDeps;
+}
+
+async function main() {
+  const packageJson = require(path.join(workspacePath, "package.json"));
+  const dependencies = Object.keys(packageJson.dependencies || {});
+  const devDependencies = Object.keys(packageJson.dependencies || {});
+  const flattenedDeps = await processDependencies(dependencies, "dependencies");
+
+  flattenedDeps.forEach(createBuildFile);
+
+  processDependencies(devDependencies, "devDependencies");
```

Comment:
We should create build files for dev dependencies too. Are there also test dependencies or other dependencies types we should handle?